src/cpp/multisolver_interface/environment.cc

```diff
@@ -0,0 +1,498 @@
+std::function<int(char* version)> XPRSgetversion = nullptr;
+std::function<int(XPRSprob prob, int attrib, int* p_value)> XPRSgetintattrib =
+    nullptr;
+/*----*/
+
+// std::function<int(XPRSprob prob, int control)> XPRSsetdefaultcontrol =
+// nullptr; std::function<int(XPRSprob prob, int control, XPRSint64 value)>
+//     XPRSsetintcontrol64 = nullptr;
+// std::function<int(XPRSprob prob, int control, const char* value)>
+//     XPRSsetstrcontrol = nullptr;
+// std::function<int(XPRSprob prob, int control, XPRSint64* p_value)>
+//     XPRSgetintcontrol64 = nullptr;
+// std::function<int(XPRSprob prob, int control, double* p_value)>
+//     XPRSgetdblcontrol = nullptr;
+// std::function<int(XPRSprob prob, int control, char* value, int maxbytes,
+//                   int* p_nbytes)>
+//     XPRSgetstringcontrol = nullptr;
+// std::function<int(XPRSprob prob, const char* probname, int ncols, int nrows,
+//                   const char rowtype[], const double rhs[], const double
+//                   rng[], const double objcoef[], const XPRSint64 start[],
+//                   const int collen[], const int rowind[],
+//                   const double rowcoef[], const double lb[], const double
+//                   ub[])>
+//     XPRSloadlp64 = nullptr;
+// std::function<int(XPRSprob prob, int row, int col, double* p_coef)>
+//     XPRSgetcoef = nullptr;
+// std::function<int(XPRSprob prob, int ncols, const int colind[])> XPRSdelcols
+// =
+//     nullptr;
+// std::function<int(XPRSprob prob, const int rowstat[], const int colstat[])>
+//     XPRSloadbasis = nullptr;
+// std::function<int(XPRSprob prob)> XPRSpostsolve = nullptr;
+// std::function<int(XPRSprob prob, char* errmsg)> XPRSgetlasterror = nullptr;
+// std::function<int(XPRSprob prob, int ncoefs, const int rowind[],
+//                   const int colind[], const double rowcoef[])>
+//     XPRSchgmcoef = nullptr;
+// std::function<int(XPRSprob prob, int nrows, const int rowind[],
+//                   const double rng[])>
+//     XPRSchgrhsrange = nullptr;
+// std::function<int(XPRSprob prob, int nrows, const int rowind[],
+//                   const char rowtype[])>
+//     XPRSchgrowtype = nullptr;
+
+bool LoadXpressFunctions(DynamicLibrary* xpress_dynamic_library) {
+  // This was generated with the parse_header_xpress.py script.
+  // See the comment at the top of the script.
+
+  // This is the 'assign' section.
+  xpress_dynamic_library->GetFunction(&XPRSinit, "XPRSinit");
+  xpress_dynamic_library->GetFunction(&XPRScopyprob, "XPRScopyprob");
+  xpress_dynamic_library->GetFunction(&XPRSwritebasis, "XPRSwritebasis");
+  xpress_dynamic_library->GetFunction(&XPRSreadprob, "XPRSreadprob");
+  xpress_dynamic_library->GetFunction(&XPRSreadbasis, "XPRSreadbasis");
+  xpress_dynamic_library->GetFunction(&XPRSgetrows, "XPRSgetrows");
+  xpress_dynamic_library->GetFunction(&XPRSgetindex, "XPRSgetindex");
+  xpress_dynamic_library->GetFunction(&XPRSgetnames, "XPRSgetnames");
```

Comment:
You can keep here only the XPRS functions that are actually called. See for example `grep XPRS`

Reply:
I commented out the unnecessary functions and added others especially for xpansion.  I have to do it also with defines.
this needs to be cleaned